downstairs/src/lib.rs

```diff
@@ -995,17 +998,17 @@ where
                                         upstairs_connection.upstairs_id,
                                 }
                             ).await?;
+                            bail!(
+                                "Upstairs connection expected \
+                                upstairs_id:{} session_id:{}  received \
+                                upstairs_id:{} session_id:{}",
+                                upstairs_connection.upstairs_id,
+                                upstairs_connection.session_id,
+                                upstairs_id,
+                                session_id
+                            );
 
-                            /*
-                             * At this point, should we just return error?
-                             * XXX
-                             */
                         } else {
-                            // matches_self above should include a check for
-                            // gen, but gen number may change between
-                            // negotiation and activation (see `XXX because the
-                            // generation number` upstairs). update generation
-                            // number here.
                             if upstairs_connection.gen != gen {
```

Comment:
do we still need this?

Reply:
Hmm..  Maybe.
If we only have `activation` and not `activation_with_gen`, then I would say yeah.
But, since I think were are heading that way, having the warning still printed would alert us that we were in this situation.